public/addEdit.js

```diff
@@ -0,0 +1,265 @@
+   dueDate.value = "";
+   tags.value = "";
+   resetSubTasks();
+   status.value = "pending";
+};
+const addNewSubTask = () => {
+   subTasksCounter = subTasksCounter + 1;
+
+   const newSubTaskDiv = document.createElement("div");
+   const label = document.createElement("label");
+   const input = document.createElement("input");
+
+   newSubTaskDiv.id = `div-sub-task-${subTasksCounter}`;
+   newSubTaskDiv.style.display = "block";
+
+   label.setAttribute("for", `sub-task-${subTasksCounter}`);
+   label.textContent = `${subTasksCounter}. `;
+
+   input.type = "text";
+   input.id = `sub-task-${subTasksCounter}`;
+
+   newSubTaskDiv.appendChild(label);
+   newSubTaskDiv.appendChild(input);
+   subTasksDiv.appendChild(newSubTaskDiv);
+   return newSubTaskDiv;
+};
+
+const deleteLastSubTask = () => {
+   subTasksDiv.removeChild(subTasksDiv.lastElementChild);
+   subTasksCounter = subTasksCounter - 1;
+};
+
+const resetSubTasks = (keepOneInput = true) => {
+   const childElementCount = subTasksDiv.childElementCount;
+   for (let index = keepOneInput ? 1 : 0; index < childElementCount; index++) {
+      deleteLastSubTask();
+   }
+   if (keepOneInput) {
+      const initialsubTask = document.getElementById("sub-task-1");
+      initialsubTask.value = "";
+   }
+};
+
+const makeSubTasksObj = () => {
+   let subTasksObj = [];
+   const inputs = subTasksDiv.getElementsByTagName("input");
+
+   for (let index = 0; index < inputs.length; index++) {
+      if (inputs[index].value != "") {
+         subTasksObj.push({
+            title: inputs[index].value,
+         });
+      }
+   }
+
+   return subTasksObj;
+};
+
+const prepareSubTasksEdit = (subTasks) => {
+   resetSubTasks(false);
+
+   for (let index = 0; index < subTasks.length; index++) {
+      let newSubTaskDiv = addNewSubTask();
+      newSubTaskDiv.getElementsByTagName("input")[0].value =
+         subTasks[index].title;
+   }
+};
+
+export const showAddEdit = async (taskId) => {
+   if (!taskId) {
+      resetFormValues();
+      addingTask.textContent = "add";
+      message.textContent = "";
+
+      setDiv(addEditDiv);
+   } else {
+      enableInput(false);
+
+      try {
+         const response = await fetch(`/api/v1/tasks/${taskId}`, {
+            method: "GET",
+            headers: {
+               "Content-Type": "application/json",
+               Authorization: `Bearer ${token}`,
+            },
+         });
+
+         const data = await response.json();
+         if (response.status === 200) {
+            prepareSubTasksEdit(data.task.subTasks);
+
+            title.value = data.task.title;
+            description.value = data.task.description;
+            isUrgent.value = data.task.isUrgent;
+            dueDate.value = data.task.dueDate;
+            tags.value = data.task.tags;
+            status.value = data.task.status;
+            addingTask.textContent = "update";
+            message.textContent = "";
+            addEditDiv.dataset.id = taskId;
```

Comment:
I am not gonna hold it up for this however, it is good practice to make functions 
 work independently (aka pure functions). In this case, your function is only able to work because they exist else were in the code. If they didn't exist, or you used this somewhere else, it would break because it is dependent on those variables. 
 
 Can you figure out a good way to pass in those references without needing to have those dependencies?